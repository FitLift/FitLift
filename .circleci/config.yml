version: 2
machine:
  timezone:
    America/Los_Angeles
jobs:
  build:
    docker:
      - image: node:9.2.0
    steps:
      - checkout
      - run: cd app && npm install
      - run: cd app && npm run test.coverage
