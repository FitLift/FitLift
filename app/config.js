export const API_URL = 'http://localhost:3000';
export const FIREBASE_URL = 'https://fitlift-38a0c.firebaseio.com';
