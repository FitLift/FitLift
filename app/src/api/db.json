{
  "currentExercises": [
    {
      "reps": 12,
      "time": 1517968200,
      "type": "Bicep Curls"
    },
    {
      "reps": 1,
      "time": 1517968200,
      "type": "Shoulder Press"
    },
    {
      "reps": 6,
      "time": 1517968200,
      "type": "Lateral Raises"
    }
  ]
}
