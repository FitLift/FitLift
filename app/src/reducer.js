import { combineReducers } from 'redux';
import navigation from './routes/reducer';
import feed from './modules/feed/redux';
import login from './modules/login/redux';
import profile from './modules/profile/redux';
import record from './modules/record/redux';

export default combineReducers({
  feed,
  login,
  navigation,
  profile,
  record,
});
