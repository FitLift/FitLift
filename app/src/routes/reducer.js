import { NavigationActions } from 'react-navigation';
import { Navigation } from './index';

// const initialState =
//   Navigation.router.getStateForAction(
//     Navigation.router.getActionForPathAndParams('login'),
//   );

const initialState =
  Navigation.router.getStateForAction(
    NavigationActions.reset({
      actions: [
        NavigationActions.navigate({ routeName: 'loggedIn' }),
      ],
      index: 0,
    }),
  );

export default (state = initialState, action) => {
  switch (action.type) {
    case 'login':
      return Navigation.router.getStateForAction(
        NavigationActions.reset({
          actions: [
            NavigationActions.navigate({ routeName: 'loggedIn' }),
          ],
          index: 0,
        }),
        state,
      );
    case 'logout':
      return initialState;
    default:
      return Navigation.router.getStateForAction(action, state);
  }
};
