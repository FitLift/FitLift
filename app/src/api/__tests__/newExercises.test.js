import nock from 'nock';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { API_URL } from '../../../config';
import { new_exercises as newExercises } from '../db.json';
import reducer, {
  TypeKeys,
  fetchNewExercises,
} from '../newExercises';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('async action creator tests', () => {
  it('should fetch current exercises', () => {
    nock(API_URL)
      .get('/new_exercises')
      .reply(200, newExercises);

    const store = mockStore();
    return store.dispatch(fetchNewExercises(5))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});


describe('reducer tests', () => {
  it('TypeKeys.RECEIVE_CURRENT_EXERCISES should work', () => {
    expect(reducer(
      {},
      {
        data: newExercises,
        type: TypeKeys.RECEIVE_NEW_EXERCISES,
      },
    )).toEqual(newExercises);
  });
});
