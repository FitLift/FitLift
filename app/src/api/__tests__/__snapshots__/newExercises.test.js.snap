// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async action creator tests should fetch current exercises 1`] = `
Array [
  Object {
    "type": "REQUEST_NEW_EXERCISES",
    "user": 5,
  },
  Object {
    "data": Array [
      Object {
        "reps": 12,
        "timeStamp": 1518407604,
        "type": "Bicep Curls",
      },
      Object {
        "reps": 5,
        "timeStamp": 1518407604,
        "type": "Shoulder Press",
      },
      Object {
        "reps": 6,
        "timeStamp": 1518407604,
        "type": "Lateral Raises",
      },
    ],
    "type": "RECEIVE_NEW_EXERCISES",
    "user": 5,
  },
]
`;
